transaction/core/src/tx.rs

```diff
@@ -401,6 +402,46 @@ impl TxOut {
             MemoPayload::default()
         }
     }
+
+    /// Creates a single output belonging to a specific recipient account.
+    /// The output is created using a predictable private key that is derived
+    /// from the input parameters.
+    ///
+    /// # Arguments:
+    /// * `recipient` - The recipient of the output.
+    /// * `domain_tag` - Domain separator for hashing the input parameters.
+    /// * `parent_block` - The parent block.
+    /// * `transactions` - The transactions that are included in the current
+    ///   block.
+    /// * `amount` - Output amount.
+    pub fn mint<T: Digestible>(
+        recipient: &PublicAddress,
+        domain_tag: &'static [u8],
+        parent_block: &Block,
+        transactions: &[T],
+        amount: Amount,
+    ) -> Result<Self, AmountError> {
+        // Create a determinstic private key based on the block contents.
```

Comment:
so I have two thoughts,

(1) I wanted to change the way this was being done in consensus enclave, because, now we are going to call this potentially twice in a block, because we will have multiple fee outputs. That means that there could be two TxOut's with the same private key. I am not sure right now that that is bad, but it seems like it could be bad, and it could be easily fixed.
(2) I feel like there is a lot of abstraction leakage here, what if instead we just took a `MerlinTranscript` as an argument and let the caller set it up how they want, and don't refer to `Block` or transactions here